jax/_src/util.py

```diff
@@ -74,6 +74,13 @@ def split_list(args, ns):
   lists.append(args)
   return lists
 
+def partition_list(bs: List[bool], l: List[Any]) -> Tuple[List[Any], List[Any]]:
+  assert len(bs) == len(l)
+  lists = lst1, lst2 = [], []  # type: ignore
+  for b, x in zip(bs, l):
+    lists[b].append(x)
+  return lst1, lst2
```

Comment:
code golf nit: this just `lists`, so you don't need the `lst1` and `lst2` binders :)